misc/python/materialize/mzcompose/services.py

```diff
@@ -239,14 +239,19 @@ def __init__(
         environment_extra: List[str] = [],
         memory: Optional[str] = None,
         options: List[str] = [],
+        scratch_directory: bool = True,
     ) -> None:
         environment = [
             "CLUSTERD_LOG_FILTER",
             "MZ_SOFT_ASSERTIONS=1",
-            "CLUSTERD_SCRATCH_DIRECTORY=/mzdata/source_data",
             *environment_extra,
         ]
 
+        if scratch_directory:
+            environment.append(
+                "CLUSTERD_SCRATCH_DIRECTORY=/mzdata/source_data",
```

Comment:
I realize this is preexisting, but `mzdata` is like the one place this _shoudn't_ go, as it survives container recreation.

```suggestion
                "CLUSTERD_SCRATCH_DIRECTORY=/scratch",
```

Reply:
moved to `/scratch` along with the necessary plumbing to make it accessible in the right places